upstream.go

```diff
@@ -697,11 +958,14 @@ func (uc *upstreamConn) handleMessage(msg *irc.Message) error {
 		}
 
 		uc.forEachDownstream(func(dc *downstreamConn) {
+			if downstreamId != 0 && downstreamId != dc.id {
+				return
+			}
```

Comment:
This piece of code used to select a specific downstream id is used multiple times throughout the code, should we create a function for that? Maybe a new `forEachDownstream` that also takes an optional downstream id.

Reply:
+1